### PR TITLE
Add option to disable autocast pass

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77566

Fix for AOT autograd, where amp has already been traced out and you don't want to re-invoke the amp pass. 

cc @anijain2305, @Chillee

### CODE REVIEW

**Review by anijain2305 — APPROVED** [review]

File: torch/csrc/jit/api/function_impl.h [hunk 1]
```diff
@@ -149,6 +155,9 @@ struct TORCH_API GraphFunction : public Function {
   // TODO: add more executors
   mutable c10::optional<ExecutorExecutionMode> executor_execution_mode_;
 
+  // if invoked on a graph that has already traced through amp
+  // don't invoke amp pass
+  mutable bool force_no_amp_ = false;
```
Comment (davidberard98) [inline torch/csrc/jit/api/function_impl.h:160]: does this need to be mutable?

File: torch/csrc/jit/python/script_init.cpp [hunk 2]
```diff
@@ -1502,6 +1502,14 @@ void initJitScriptBindings(PyObject* module) {
       .def_property_readonly(
           "name",
           [](const StrongFunctionPtr& self) { return self.function_->name(); })
+      .def(
+          "_set_ignore_amp",
```
Comment (davidberard98) [inline torch/csrc/jit/python/script_init.cpp:1506]: do we need anything like this for modules?

**Review by davidberard98 — APPROVED** [review]
had some small comments, otherwise looks good

File: torch/csrc/jit/python/script_init.cpp (hunk 2 above)
Comment (eellison) [reply torch/csrc/jit/python/script_init.cpp:1506]: no, since the use case here is just aot autograd

**Pull request closed without merging**